virtcontainers/container.go

```diff
@@ -1296,8 +1296,90 @@ func (c *Container) updateResources(oldResources, newResources ContainerResource
 		// recalculate the actual number of vCPUs if a different number of vCPUs was removed
 		newResources.VCPUs = oldVCPUs - vcpusRemoved
 	}
+	return nil
+}
+
+func (c *Container) checkMemHotplugValid(mem *uint32) error {
+	state, err := c.sandbox.storage.fetchSandboxState(c.sandbox.id)
+	if err != nil {
+		return fmt.Errorf("Cound not get the state of sandbox")
+	}
+
+	memorySectionSizeMB := state.GuestMemoryBlockSizeMB
+	if memorySectionSizeMB == 0 {
+		return nil
+	}
+
+	// TODO: hot add memory aligned to memory section should be more properly. See https://github.com/kata-containers/runtime/pull/624#issuecomment-419656853
+	*mem = uint32(math.Ceil(float64(*mem)/float64(memorySectionSizeMB))) * memorySectionSizeMB
```

Comment:
I wonder if it is proper to change mem in this function because it says "check" . You can add newmem as a return value or rename this function.

Reply:
Ok, I think add newmem as a return value will be more proper.

Reply:
updated